sky/utils/resources_utils.py

```diff
@@ -0,0 +1,63 @@
+"""Resources related utilities."""
+from sky.clouds import service_catalog
+from sky.utils import ux_utils
+
+# List of all accelerators supported by SkyPilot.
+# NOTE: Must include accelerators supported for local clusters.
+_ACCELERATORS = [
+    'A100',
+    'A10G',
+    'Gaudi HL-205',
+    'Inferentia',
+    'K520',
+    'K80',
+    'M60',
+    'Radeon Pro V520',
+    'T4',
+    'T4g',
+    'V100',
+    'V100-32GB',
+    'Virtex UltraScale (VU9P)',
+    'A10',
+    'A100-80GB',
+    'P100',
+    'P40',
+    'Radeon MI25',
+    'P4',
+]
+
+
+def canonicalize_accelerator_name(accelerator: str) -> str:
+    """Returns the canonical accelerator name."""
+    # TPU names are always lowercase.
+    if accelerator.lower().startswith('tpu-'):
+        return accelerator.lower()
+
+    mapping = {name.lower(): name for name in _ACCELERATORS}
+    if accelerator.lower() in mapping:
+        return mapping[accelerator.lower()]
+
+    # A new accelerator can be added after upgrading SkyPilot.
+    # Search the accelerator name in the service catalog.
+    searched = service_catalog.list_accelerators(name_filter=accelerator,
+                                                 case_sensitive=False)
+    names = list(searched.keys())
+
+    # Exact match.
+    if accelerator in names:
+        return accelerator
+
+    if len(names) == 0:
+        with ux_utils.print_exception_no_traceback():
+            raise ValueError(f'Unknown accelerator: {accelerator}')
```

Comment:
UX: 
```
(py37) ~ » sky launch --gpus V1000 ''                                                              2 ↵
I 08-14 08:29:48 optimizer.py:865] No resource satisfying {'V1000': 1} on [AWS, Azure, GCP].
sky.exceptions.ResourcesUnavailableError: No launchable resource found for task sky-cmd. To fix: relax its resource requirements.
Hint: 'sky show-gpus --all' to list available accelerators.
      'sky check' to check the enabled clouds.

(py37) ~ » sky launch --gpus V1000 ''                                                              1 ↵
ValueError: Unknown accelerator: V1000
```
Top is master branch's error message, bottom is this branch.  I feel like the former message is more informative. Can we fall back to that somehow, or move that to here?

Reply:
Good catch. The error will be handled by the optimizer.